source/tutorials/nixos-configuration-on-vm.md

```diff
@@ -0,0 +1,205 @@
+On NixOS your configuration generated using `nix-generate-config` contains this user configuration commented out.
+:::
+
+Additionally, you need to specify a password for this user.
+For the purpose of demonstration only, you specify an insecure, plain text password by adding the `initialPassword` option to the user configuration:[^password]
+
+[^password]: Warning: Do not use plain text passwords outside of this example unless you know what you are doing. See [`initialHashedPassword`](https://nixos.org/manual/nixos/stable/options.html#opt-users.extraUsers._name_.initialHashedPassword) or [`ssh.authorizedKeys`](https://nixos.org/manual/nixos/stable/options.html#opt-users.extraUsers._name_.openssh.authorizedKeys.keys) for more secure alternatives.
+
+```nix
+initialPassword = "testpw";
+```
+
+This tutorial focuses on testing NixOS configurations on a virtual machine.
+Therefore you will remove the reference to `hardware-configuration.nix`:
+
+```diff
+-  imports =  [ ./hardware-configuration.nix ];
+```
+
+The complete `configuration.nix` file now looks like this:
+
+```nix
+{ config, pkgs, ... }:
+{
+  boot.loader.systemd-boot.enable = true;
+  boot.loader.efi.canTouchEfiVariables = true;
+
+  services.xserver.enable = true;
+
+  services.xserver.displayManager.gdm.enable = true;
+  services.xserver.desktopManager.gnome.enable = true;
+
+  users.users.alice = {
+    isNormalUser = true;
+    extraGroups = [ "wheel" ]; # Enable ‘sudo’ for the user.
+    packages = with pkgs; [
+      firefox
+      tree
+    ];
+    initialPassword = "testpw";
+  };
+
+  system.stateVersion = "22.11";
+}
+```
+
+## Creating a QEMU based virtual machine using a configuration
+
+A virtual machine is created with the `nix-build` command.
+
+To select `configuration.nix` in the working directory, specify the configuration file on the command line:
+
+```shell-session
+nix-build '<nixpkgs/nixos>' -A vm \
+-I nixpkgs=channel:nixos-22.11 \
+-I nixos-config=./configuration.nix
+```
+
+This command builds the attribute `vm` from the `nixos-22.11` release of NixOS, using the NixOS configuration as specified in the relative path.
+
+<details><summary> Detailed explanation </summary>
+
+The first optional argument of [nix-build](https://nixos.org/manual/nix/stable/command-ref/nix-build.html) is a path to the derivation to be build.
+With `'<nixpkgs>'` Nix is instructed to resolve the search path defined with the [`NIX_PATH` environment variable](https://nixos.org/manual/nix/stable/command-ref/env-common.html#env-NIX_PATH) or the [`-I` option](https://nixos.org/manual/nix/unstable/command-ref/opt-common.html#opt-I).
+The virtual machine setup is provided by NixOS, which is part of the `nixpkgs` repository, therefore we use `'<nixpkgs/nixos>'`.
+The [-A option](https://nixos.org/manual/nix/stable/command-ref/opt-common.html#opt-attr) specifies the attribute to pick from the provided [Nix expression `<nixpkgs>`](https://nix.dev/tutorials/nix-language#search-path).
```

Comment:
```suggestion
The [`-A` option](https://nixos.org/manual/nix/stable/command-ref/opt-common.html#opt-attr) specifies the attribute to pick from the provided [Nix expression `<nixpkgs>`](https://nix.dev/tutorials/nix-language#search-path).
```

No idea how to do that [correctly with MyST](https://myst-parser.readthedocs.io/en/latest/syntax/cross-referencing.html#reference-roles), but this should not be a plain URL to nix.dev. Let Sphinx do the heavy lifting.

Reply:
I only get myst links working by adding a explicit target to the nix-language tutorial like nix-tut-search-path because the heading is the same as in anti-patterns/language.md

Reply:
That's fine, `search-path-tutorial` is probably good. Global namespaces should be forbidden by law, sigh.